pkg/api/operators/apis/experiment/v1alpha2/types.go

```diff
@@ -0,0 +1,224 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha2
+
+import (
+	pb "github.com/kubeflow/katib/pkg/api"
+	trial "github.com/kubeflow/katib/pkg/api/operators/apis/trial/v1alpha2"
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type ExperimentSpec struct {
+	// List of hyperparameter configurations.
+	Parameters []ParameterSpec `json:"parameters,omitempty"`
+
+	// Describes the objective of the experiment.
+	Objective *ObjectiveSpec `json:"objective,omitempty"`
+
+	// Describes the suggestion algorithm.
+	Algorithm *AlgorithmSpec `json:"algorithm,omitempty"`
+
+	// Template for each run of the trial.
+	TrialTemplate *TrialTemplate `json:"trialTemplate,omitempty"`
+
+	// How many trials can be processed in parallel.
+	ParallelTrialCount int `json:"parallelTrialCount,omitempty"`
+
+	// Total number of trials to run.
+	MaxTrialCount      int `json:"maxTrialCount,omitempty"`
+
+	// TODO - figure out what to do with metric collectors
+	MetricsCollectorType string `json:"metricsCollectorSpec,omitempty"`
+
+	// TODO - Other fields, exact format is TBD. Will add these back during implementation.
+	// - NAS
+	// - Early stopping
+	// - Resume experiment
+}
+
+type ExperimentStatus struct {
+	// Represents time when the Experiment was acknowledged by the Experiment controller.
+	// It is not guaranteed to be set in happens-before order across separate operations.
+	// It is represented in RFC3339 form and is in UTC.
+	StartTime         *metav1.Time `json:"startTime,omitempty"`
+
+	// Represents time when the Experiment was completed. It is not guaranteed to
+	// be set in happens-before order across separate operations.
+	// It is represented in RFC3339 form and is in UTC.
+	CompletionTime    *metav1.Time `json:"completionTime,omitempty"`
+
+	// Represents last time when the Experiment was reconciled. It is not guaranteed to
+	// be set in happens-before order across separate operations.
+	// It is represented in RFC3339 form and is in UTC.
+	LastReconcileTime *metav1.Time `json:"lastReconcileTime,omitempty"`
+
+	// List of observed runtime conditions for this Experiment.
+	Conditions []ExperimentCondition `json:"conditions,omitempty"`
+
+	// Current optimal trial parameters and observations.
+	CurrentOptimalTrial OptimalTrial `json:"currentOptimalTrial,omitempty"`
+
+	// How many trials have successfully completed.
+	TrialsCompleted int `json:"trialsCompleted,omitempty"`
+
+	// How many trials have failed.
+	TrialsFailed int `json:"trialsFailed,omitempty"`
+
+	// How many trials have been killed.
+	TrialsKilled int `json:"trialsKilled,omitempty"`
+
+	// How many trials are currently pending.
+	TrialsPending int `json:"trialsPending,omitempty"`
+}
+
+type OptimalTrial struct {
+	// Key-value pairs for the current optimal hyperparameters.
+	Hyperparameters []trial.Hyperparameter `json:"hyperparameters"`
```

Comment:
Is this the same as `Assignments`? Should that go on the `Observation`?

Reply:
It is the same as Assignments. I think it should be outside of Observation since the two are parallel concepts (input and output basically).